### ISSUE TITLE
Space around keyword produces misleading results for super

### ISSUE THREAD

**Issue by rwz** [bbatsov/rubocop#3116]
```ruby
def my_method
  super&.a_method_on_super
end
```

This code produces an offense of `Style/SpaceAroundKeyword`: Space after keyword super is missing.

Seems like a false positive to me.

**Comment by owst** [comment]
Can you give a minimal reproducible example? I tried your code, but contains a syntax error:
```
> cat foo.rb
def my_method
  super&.a_method_on_super
end
> ruby foo.rb
foo.rb:2: syntax error, unexpected '.'
  super&.a_method_on_super
         ^
```

So I removed the `&`, but that code gives no error for me:
```
> rubocop foo.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

I'm running `rubocop`: `0.39.0 (using Parser 2.3.1.0, running on ruby 2.2.3 x86_64-darwin14)`.

Can you provide the output of running `rubocop -V`?

**Comment by rwz** [comment]
The code sample uses ruby 2.3 safe navigation, so you need at least 2.3.0 for it to work.

The rubocop version is 0.39.0

**Comment by rwz** [comment]
Minimal example using ruby 2.3:

```ruby
# foo.rb
def my_method
  super&.a_method_on_super
end
```

```yaml
# .rubocop.yml
Style/SpaceAroundKeyword:
  Enabled: true
```
```sh
$ rubocop -D foo.rb
Inspecting 1 file
E

Offenses:

foo.rb:2:8: E: unexpected token error
(Using Ruby 2.0 parser; configure using TargetRubyVersion parameter, under AllCops)
  super&.a_method_on_super
       ^^

1 file inspected, 1 offense detected
```

**Comment by rwz** [comment]
As seen from the example, updating to 0.40 didn't change much.

**Comment by segiddins** [comment]
I have a fix for this locally, about to make a PR

**Comment by owst** [comment]
Aha, of course, I forgot about the new safe navigation operator (I'm stuck on `2.2.3` at work, so haven't used it)! Thanks for adding the example @rwz, sounds like a fix is incoming shortly.

**Comment by tejasbubane** [comment]
Fixed in #3118. This can be closed @rwz 

@segiddins Going further please add something like `fixes #<issue_number>` to your PR description so that it automatically closes the associated issue. Read more about it [here](https://github.com/blog/1506-closing-issues-via-pull-requests).

**Comment by rwz** [comment]
Cool, thanks for taking care of it! :+1:
